### PR TITLE
V0.8.0 variousfixes

### PR DESCRIPTION
**Pull request by miskreant**
## Description
Various minor fixes for 0.8.0 beta, putting up for feedback / guidance.

## Motivation and Context

- Collapsable rows would not expand
- Reorderable columns, no UX positioning line, drag behavior broken
- Resizable columns, no UX positioning line
- Vertical scrollbar click did not work

## How Has This Been Tested?
Manual testing, still work to be done wrt client side filtering (duplicated rows when fewer rows than table height)

## Screenshots (if appropriate):
https://drive.google.com/file/d/0B6R1mUc10dJfNmUwUEdia2xOM2s/view?usp=sharing

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.

### CODE REVIEW

File: src/FixedDataTable.js [hunk 1]
```diff
@@ -639,7 +640,8 @@ var FixedDataTable = React.createClass({
   },
 
   _renderRows(/*number*/ offsetTop) /*object*/ {
-    var state = this.state;
+    let state = updateRowHeights(this.state, this.props);
+    state = updateVisibleRows(state);
```
Comment (miskreant) [inline src/FixedDataTable.js]: Collapsable rows need to re-calculate the row heights as part of the re-render. I'm not sure if hijacking from the scrollStateHelper like this is a good approach though.. thoughts?

File: src/FixedDataTable.js [hunk 2]
```diff
@@ -905,8 +907,7 @@ var FixedDataTable = React.createClass({
       if (!this._isScrolling) {
         this._didScrollStart();
       }
-
-      this.scrollActions.scrollToY(scrollPos);
+      this.props.scrollActions.scrollToY(scrollPos);
```
Comment (miskreant) [inline src/FixedDataTable.js]: ScrollActions now a prop

File: src/FixedDataTableCellGroup.js [hunk 3]
```diff
@@ -80,8 +80,8 @@ var FixedDataTableCellGroupImpl = React.createClass({
     for (var i = 0, j = columns.length; i < j; i++) {
       var columnProps = columns[i].props;
       var cellTemplate = columns[i].template;
-      var recycable = columnProps.allowCellsRecycling && !isColumnReordering;
-      if (!recycable || (
+      var recyclable = columnProps.allowCellsRecycling && !isColumnReordering;
+      if (!recyclable || (
```
Comment (miskreant) [inline src/FixedDataTableCellGroup.js]: minor nit

File: src/FixedDataTableColumnReorderHandle.js [hunk 4]
```diff
@@ -80,7 +80,7 @@ var FixedDataTableColumnReorderHandle = React.createClass({
   onMouseDown(event) {
     var targetRect = event.target.getBoundingClientRect();
 
-    var mouseLocationInElement = event.clientX - targetRect.offsetLeft;
+    var mouseLocationInElement = event.clientX - targetRect.left;
```
Comment (miskreant) [inline src/FixedDataTableColumnReorderHandle.js]: offsetLeft was NaN

File: src/FixedDataTableContainer.js [hunk 5]
```diff
@@ -53,8 +53,8 @@ export default class FixedDataTableContainer extends React.Component {
   render() {
     return (
       <FixedDataTable 
-        {...this.state} 
         {...this.props} 
+        {...this.state}
```
Comment (miskreant) [inline src/FixedDataTableContainer.js]: Props was overriding state, broke reordering

File: src/FixedDataTable.js (hunk 1 above)
Comment (KamranAsif) [inline src/FixedDataTable.js]: Woh, we definitely don't want to call this here. They are expensive operations. 
We need to track what went wrong in the reducer and add correct paths there.

Comment (miskreant) [inline src/FixedDataTable.js]: Aye, thx for the tip

Comment (wcjordan) [inline src/FixedDataTable.js]: Ya I wonder if we need to provide a hook for the user to invalidate row heights when something changes...  And if so do we want to invalidate all of them or just a subset?

**Comment by wcjordan** [comment]
Looks good to me except for the row height issue.  Maybe we can remove that piece, merge this in, and then address that in a future review after some discussion of the best approach.  @miskreant did you determine why the row height change was getting picked up on master but not on this branch?

Also in the video, column reordering looks slow.  I think that's somewhat common for dev mode, could you test in the prod build and see if it's smooth?

**Comment by wcjordan** [comment]
lgtm.  is this ready to merge?

**Comment by miskreant** [comment]
Ya, these can go in.

WRT your question about why row height is picked up on master but not here, on master the row re-rendering is built right into FixedDataTableBufferedRows#render, but on 0.8.0 we've moved away from that.

**Comment by miskreant** [comment]
@wcjordan And column reordering is snappy in prod, tested in LD

**Pull request merged**